gtda/images/filtrations.py

```diff
@@ -490,3 +490,143 @@ def transform(self, X, y=None):
         Xt = np.concatenate(Xt)
 
         return Xt
+
+
+@adapt_fit_transform_docs
+class ErosionFiltration(BaseEstimator, TransformerMixin):
+    """Filtrations of 2D/3D binary images based on the erosion of activated
+    regions.
+
+    This filtration assigns to each pixel in an image a grayscale value
+    calculated as follows. If the minimum Manhattan distance between the
+    pixel and any deactivated pixel in the image is less than or equal to
+    the parameter `n_iterations`, the assigned value is this distance –
+    in particular, deactivated pixels are assigned a value of 0.
+    Otherwise, the assigned grayscale value is the sum of the lengths
+    along all axes of the image – equivalently, it is the maximum
+    Manhattan distance between any two pixels in the image. The name of
+    this filtration comes from the fact that these values can be computed
+    by iteratively eroding activated regions, shrinking them by a total
+    amount `n_iterations`.
```

Comment:
Could you mention `erosion` as a morphological operator?

Reply:
I can mention it but I really don't think that it is worth creating a `MorphologicalFiltration`. This would add too much complexity for the user. The same logic could be applied to persistent homology modules. We could have only one class taking a string `filtration` that could be `rips`, `sparse_rips`, `cech`, `alpha`. I think it's much easier to describe clearly all of them separately in the documentation

Indeed, one can be obtained from the other by inverting the binary images. One could be implemented from the other but I think its better to have both of them and simply extract what they share (the dilation function) and put it in a `_utils.py` as you suggested.

Reply:
Sure, I will mention it for dilation too !